### PR TITLE
ref(pagerduty): Add service table

### PR DESCRIPTION
**Pull request by MeredithAnya**
Screen Shot 2019-11-05 at 11 31 11 AM

*more details incoming...*

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+     * the columns to appear - order doesn't matter in columnLabels
+     */
+    columnKeys: PropTypes.arrayOf(PropTypes.string),
+    /**
+     * Automatically save even if fields are empty
+     */
+    allowEmpty: PropTypes.bool,
+  };
+
+  static defaultProps = {
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx]: @MeredithAnya I recommend using the pattern of using the types of `defaultProps` to set the type of your `props`. See `/components/events/eventDataSection.tsx` as an example.

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    };
+
+    const removeRow = rowIndex => {
+      //eslint-disable-next-line no-unused-vars
+      const newValue = [...value];
+      newValue.splice(rowIndex, 1);
+      saveChanges(newValue);
+    };
+
+    const setValue = (rowIndex: number, fieldKey: string, fieldValue: any) => {
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx]: @MeredithAnya Can we do better than `any` for `fieldValue`?

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            </Flex>
+          ))}
+        </Flex>
+        {value.map((row, rowIndex) => (
+          <Flex key={rowIndex} align="center" mt={1}>
+            {mappedKeys.map((fieldKey: string, i: number) => (
+              <Flex key={fieldKey} align="center" ml={1} flex="1 0 0">
+                <Box flex={1}>
+                  <Input
+                    onChange={(v: string) => setValue(rowIndex, fieldKey, v ? v : null)}
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx]: @MeredithAnya I am confused about this:
1. Why do we need the ternary statement `v ? v : null`?
2. I thought `v` has the property `currentTarget`? How could it be a string here?

File: src/sentry/integrations/pagerduty/integration.py [hunk 4]
```diff
@@ -92,23 +134,34 @@ def build_integration(self, state):
...
 
     def dispatch(self, request, pipeline):
         if "config" in request.GET:
             pipeline.bind_state("config", request.GET["config"])
             return pipeline.next_step()
 
-        return self.redirect(self.get_app_url())
+        account_name = None
+        if "account" in request.GET:
+            account_name = request.GET["account"]
```
Comment (scefali) [inline src/sentry/integrations/pagerduty/integration.py]: @MeredithAnya would it make more sense to use `getattr` here instead?

File: src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx [hunk 5]
```diff
@@ -52,6 +54,35 @@ class ConfigureIntegration extends AsyncView<Props, State> {
...
   }
 
+  onUpdateIntegration = () => {
+    this.setState(this.getDefaultState(), this.fetchData);
+  };
+
+  getAction = (provider: IntegrationProvider | undefined) => {
+    const {integration} = this.state;
+    const action =
+      provider && provider.key === 'pagerduty' ? (
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx]: :(

Comment (MeredithAnya) [reply src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx]: yeahhh..... :/

File: src/sentry/integrations/pagerduty/integration.py [hunk 6]
```diff
@@ -92,23 +134,34 @@ def build_integration(self, state):
...
 
 class PagerDutyInstallationRedirect(PipelineView):
-    def get_app_url(self):
+    def get_app_url(self, account_name=None):
+        if not account_name:
+            account_name = "app"
+
         app_id = options.get("pagerduty.app-id")
         setup_url = absolute_uri("/extensions/pagerduty/setup/")
+        setup_url = "https://meredith.ngrok.io/extensions/pagerduty/setup/"
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/integrations/pagerduty/integration.py]: yeah this should have been left out my b

**Review by scefali — APPROVED** [review]

**Pull request merged**